roboflow/core/version.py

```diff
@@ -318,18 +318,32 @@ def deploy(self, model_type: str, model_path: str) -> None:
             class_names.sort(key=lambda x: x[0])
             class_names = [x[1] for x in class_names]
 
-            model_artifacts = {
-                "names": class_names,
-                "yaml": model["model"].yaml,
-                "nc": model["model"].nc,
-                "args": {
-                    k: val
-                    for k, val in model["model"].args.__dict__.items()
-                    if k != "hydra"
-                },
-                "ultralytics_version": ultralytics.__version__,
-                "model_type": model_type,
-            }
+            try:
+                model_artifacts = {
+                    "names": class_names,
+                    "yaml": model["model"].yaml,
+                    "nc": model["model"].nc,
+                    "args": {
+                        k: val
+                        for k, val in model["model"].args.items()
+                        if ((k == "model") or (k == "imgsz") or (k == "batch"))
+                    },
+                    "ultralytics_version": ultralytics.__version__,
+                    "model_type": model_type,
+                }
+            except:
```

Comment:
if we knew the exception thrown then some other error will not get caught and handled with this catch, and you could have a "catch all" for that case